lib/buffer.js

```diff
@@ -1141,10 +1142,10 @@ Buffer.prototype.write = function write(string, offset, length, encoding) {
 
 Buffer.prototype.toJSON = function toJSON() {
   if (this.length > 0) {
-    const data = new Array(this.length);
-    for (let i = 0; i < this.length; ++i)
-      data[i] = this[i];
-    return { type: 'Buffer', data };
+    return {
+      type: 'Buffer',
+      data: ArrayFrom(new Array(this.length), (_value, i) => this[i]),
```

Comment:
This was the fastest [when I last benched](https://perf.link/#eyJpZCI6ImFqZXBrM3E0dnAxIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMTAwMCkua2V5cygpXSIsInRlc3RzIjpbeyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6Im5ldyBBcnJheShkYXRhLmxlbmd0aCkuZmlsbCgpLm1hcCgoXywgaSkgPT4gZGF0YVtpXSkiLCJydW5zIjpbMTEwMDAsMTEwMDAsNDAwMCwxMDAwLDI0MDAwLDcwMDAsMTAwMCwxMDAwLDEwMDAsMzcwMDAsNTAwMCwyMjAwMCw0MjAwMCwxMDAwMCwxMTAwMCwzNjAwMCw0MDAwLDYwMDAsNjAwMDAsNDUwMDAsNjAwMCw0MzAwMCw2NTAwMCw0MjAwMCw1NzAwMCwyOTAwMCwxMDAwLDUxMDAwLDEwMDAsODAwMCwxMDAwLDY3MDAwLDYxMDAwLDE4MDAwLDEwMDAsMTEwMDAsNDcwMDAsMjgwMDAsNDIwMDAsMzAwMDAsNDkwMDAsNTQwMDAsMTYwMDAsNjQwMDAsNTEwMDAsMzgwMDAsNjkwMDAsNTMwMDAsNDEwMDAsMzAwMDAsMjcwMDAsNjkwMDAsNjEwMDAsNTQwMDAsNzAwMCwxODAwMCwzNzAwMCw0NzAwMCwxNTAwMCwxMDAwLDExMDAwLDYxMDAwLDI5MDAwLDQ1MDAwLDUzMDAwLDQxMDAwLDQ4MDAwLDI5MDAwLDYxMDAwLDM3MDAwLDI4MDAwLDUwMDAsMTAwMCw3MTAwMCwyNTAwMCw0ODAwMCw2ODAwMCw0MzAwMCw2MjAwMCwxMTAwMCw2NzAwMCw1ODAwMCw1MjAwMCwxNzAwMCw3MTAwMCw1NjAwMCw0OTAwMCw2MDAwLDY2MDAwLDMwMDAwLDE1MDAwLDEwMDAsNTAwMCwxMDAwLDU1MDAwLDYyMDAwLDQwMDAsMzAwMCwxMDAwLDY1MDAwXSwib3BzIjozMjEwMH0seyJuYW1lIjoiRmluZCBpdGVtIDEwMCIsImNvZGUiOiJBcnJheS5mcm9tKG5ldyBBcnJheShkYXRhLmxlbmd0aCksIChfLCBpKSA9PiBkYXRhW2ldKSIsInJ1bnMiOls2MDAwLDYwMDAsMzAwMCwxMDAwLDEyMDAwLDQwMDAsMTAwMCwxMDAwLDEwMDAsMjAwMDAsNDAwMCwxMTAwMCw4MDAwLDcwMDAsNjAwMCwxODAwMCwzMDAwLDUwMDAsMjUwMDAsMjQwMDAsNTAwMCwxNzAwMCwyNDAwMCwxMzAwMCwyMzAwMCwxMzAwMCwyNDAwMCwxMDAwLDIzMDAwLDYwMDAsMjQwMDAsMjQwMDAsMjQwMDAsOTAwMCwxMDAwLDYwMDAsMjEwMDAsMTcwMDAsMTIwMDAsMTYwMDAsMTMwMDAsMTEwMDAsOTAwMCwyMTAwMCwxOTAwMCwxNzAwMCwyNTAwMCwxNTAwMCwyNDAwMCwyNDAwMCwxMTAwMCwyMjAwMCwyNDAwMCwyNDAwMCw1MDAwLDEzMDAwLDExMDAwLDIzMDAwLDcwMDAsMTAwMCw2MDAwLDI1MDAwLDEyMDAwLDIwMDAwLDI0MDAwLDE4MDAwLDI0MDAwLDExMDAwLDI0MDAwLDE2MDAwLDI0MDAwLDQwMDAsMjQwMDAsMjYwMDAsMTMwMDAsMTAwMCwyNjAwMCwyMjAwMCwyNTAwMCwxMDAwMCwyNTAwMCwyNDAwMCwyNDAwMCw2MDAwLDI2MDAwLDIzMDAwLDIzMDAwLDIwMDAsMjQwMDAsMTYwMDAsMTIwMDAsMjEwMDAsNDAwMCwxMDAwLDI0MDAwLDI0MDAwLDI0MDAwLDIwMDAsMTAwMCwxODAwMF0sIm9wcyI6MTQ2NzB9LHsibmFtZSI6IkZpbmQgaXRlbSA4MDAiLCJjb2RlIjoiQXJyYXkuZnJvbSh7IGxlbmd0aDogZGF0YS5sZW5ndGggfSwgKF8sIGkpID0%2BIGRhdGFbaV0pIiwicnVucyI6WzUwMDAsNTAwMCwzMDAwLDEwMDAsNjAwMCw3MDAwLDkwMDAsMTAwMCwxMDAwLDEwMDAwLDIwMDAsODAwMCwxMDAwLDYwMDAsNTAwMCw3MDAwLDE3MDAwLDQwMDAsMTkwMDAsMTUwMDAsNDAwMCw4MDAwLDE4MDAwLDE0MDAwLDgwMDAsNjAwMCwxMTAwMCwxOTAwMCwxNDAwMCw0MDAwLDE0MDAwLDE2MDAwLDIwMDAwLDcwMDAsMTkwMDAsNTAwMCwxNTAwMCw3MDAwLDkwMDAsNzAwMCw1MDAwLDEyMDAwLDYwMDAsMTUwMDAsMTAwMDAsMTIwMDAsMjQwMDAsMTUwMDAsMTcwMDAsMTMwMDAsODAwMCwyMjAwMCwxOTAwMCwxOTAwMCw0MDAwLDEyMDAwLDgwMDAsMTYwMDAsNDAwMCwxNTAwMCw1MDAwLDIxMDAwLDkwMDAsMTQwMDAsMTMwMDAsMTIwMDAsMTAwMDAsODAwMCwxNDAwMCw5MDAwLDIxMDAwLDMwMDAsMjAwMDAsMjMwMDAsMTMwMDAsOTAwMCwyNDAwMCwxNTAwMCwyMTAwMCw4MDAwLDIzMDAwLDE5MDAwLDE4MDAwLDcwMDAsMjQwMDAsMTgwMDAsMTAwMDAsNDAwMCwxOTAwMCwxMzAwMCw3MDAwLDEwMDAsNDAwMCwxMDAwLDE5MDAwLDE2MDAwLDE2MDAwLDIwMDAsMTAwMCwxNjAwMF0sIm9wcyI6MTEyMzB9XSwidXBkYXRlZCI6IjIwMjQtMDktMDdUMjI6MDg6NTEuMjEyWiJ9)

```suggestion
      data: new Array(this.length).fill().map((_, i) => this[i]),
```

Reply:
But yeah not sure with the primordials

Reply:
Primordial wise, that code would become

```suggestion
      data: ArrayPrototypeMap(ArrayPrototypeFill(new Array(this.length)), (_, i) => this[i]));
```

Which probably wouldn't be faster. However, `ArrayOf` may be.

Reply:
> Would ArrayOf(...this) work? Not sure about speed tho

Using spread operator on a function like array of would hit the limits of v8 argument length limit which is 2^32 -1.